mne/viz/evoked.py

```diff
@@ -1529,162 +1686,203 @@ def plot_compare_evokeds(evokeds, picks=list(), gfp=False, colors=None,
+        channel types, a list of figures, one for each channel type is
+        returned.
+
+    Notes
+    -----
+    When multiple channels are passed, this function combines them all, to
+    get one time course for each condition. If gfp is True it combines
+    channels using global field power (GFP) computation, else it is taking
+    a plain mean.
+
+    This function is useful for comparing multiple ER[P/F]s - e.g., for
+    multiple conditions - at a specific location.
+
+    It can plot:
+
+    - a simple :class:`mne.Evoked` object,
+    - a list or dict of :class:`mne.Evoked` objects (e.g., for multiple
+      conditions),
+    - a list or dict of lists of :class:`mne.Evoked` (e.g., for multiple
+      subjects in multiple conditions).
+
+    In the last case, it can show a confidence interval (across e.g. subjects)
+    using parametric or bootstrap estimation.
+
+    When ``picks`` includes more than one planar gradiometer, the planar
+    gradiometers are combined with RMSE. For example data from a
+    VectorView system with 204 gradiometers will be transformed to
+    102 channels.
     """
     import matplotlib.pyplot as plt
-    from ..evoked import Evoked, combine_evoked
+    import matplotlib.lines as mlines
 
-    # set up labels and instances
-    if isinstance(evokeds, Evoked):
-        evokeds = dict(Evoked=evokeds)  # title becomes 'Evoked'
-    elif not isinstance(evokeds, dict):
-        evokeds = dict((str(ii + 1), evoked)
-                       for ii, evoked in enumerate(evokeds))
-    for cond in evokeds.keys():
-        if not isinstance(cond, string_types):
-            raise TypeError('Conditions must be str, not %s' % (type(cond),))
+    evokeds, colors = _format_evokeds_colors(evokeds, cmap, colors)
     conditions = sorted(list(evokeds.keys()))
 
+    # check ci parameter
+    if ci is None:
+        ci = False
+    if ci is True:
+        ci = .95
+    elif ci is not False and not (isinstance(ci, np.float) or callable(ci)):
+        raise TypeError('ci must be None, bool, float or callable, got %s' %
+                        type(ci))
+
     # get and set a few limits and variables (times, channels, units)
-    example = (evokeds[conditions[0]]
-               if isinstance(evokeds[conditions[0]], Evoked)
-               else evokeds[conditions[0]][0])
-    if not isinstance(example, Evoked):
-        raise ValueError("evokeds must be an instance of mne.Evoked "
-                         "or a collection of mne.Evoked's")
-    times = example.times
+    one_evoked = evokeds[conditions[0]][0]
+    times = one_evoked.times
+    info = one_evoked.info
+    ch_names = one_evoked.ch_names
     tmin, tmax = times[0], times[-1]
-    if (tmin >= 0 or tmax <= 0) and vlines == [0.]:
-        vlines = list()
+
+    if vlines == "auto" and (tmin < 0 and tmax > 0):
+        vlines = [0.]
+    if not isinstance(vlines, (list, tuple)):
+        raise TypeError("vlines must be a list or tuple, not " + type(vlines))
```

Comment:
This must not be covered by a test because it will error, you should use `%` as we usually do:
```
In [1]: 'a' + type('b')
Traceback (most recent call last):

  File "<ipython-input-1-03514d031c8f>", line 1, in <module>
    'a' + type('b')

TypeError: must be str, not type
```